packages/mui-utils/src/index.ts

```diff
@@ -25,6 +25,9 @@ export { default as unstable_unsupportedProp } from './unsupportedProp';
 export { default as unstable_useControlled } from './useControlled';
 export { default as unstable_useEventCallback } from './useEventCallback';
 export { default as unstable_useForkRef } from './useForkRef';
+export { default as unstable_useLazyRef } from './useLazyRef';
+export { default as unstable_useTimeout, Timeout as unstable_Timeout } from './useTimeout';
+export { default as unstable_useOnMount } from './useOnMount';
```

Comment:
I would have released these without the `unstable_` prefix but I see that all your hooks use this format so I've kept them consistent with that.

Reply:
We had this whole conversation about how we want to treat this package. We used `unstable_` to allow ourselves to do breaking changes whenever we need. There were discussions that we may want to keep this package in alpha all the time, this would mean that Base UI, Material UI and other packages will re-export the things that makes sense, or make sure the APIs are consistent. @ would tag @michaldudak as a owner of the @mui/utils package to share his opinion.

Reply:
Is this package used externally? It doesn't seem to be publicized, doesn't have a readme on npm and doesn't seem to have docs. If it's not used externally, there shouldn't be any downsides to publishing a new major version each time we need to make changes. I must say it's a bit annoying to have to destructure the `unstable_` prefix each time it's used in the datagrid.

For the hooks that I've introduced here, I also don't see any potential breaking change in the future. Their behavior is pretty clear and fundamental, e.g. I don't see how `useLazyRef` could be expanded in a breaking way.

Reply:
I added it for discussion on Monday's Core weekly. Let us know if you wish to attend to discuss this @romgrk 😊